cmd/outline-ss-server/metrics.go

```diff
@@ -49,11 +61,100 @@ type outlineMetrics struct {
 var _ service.TCPMetrics = (*outlineMetrics)(nil)
 var _ service.UDPMetrics = (*outlineMetrics)(nil)
 
+type activeClient struct {
+	IPKey           IPKey
+	connectionCount int
+	startTime       time.Time
+}
+
+func (c *activeClient) IsActive() bool {
```

Comment:
I don't think this is helping readability, since it forced me to look at the definition at the call site. Just inlining the code is probably more readable.